cmd/server/pactasrv/initiative_invitation.go

```diff
@@ -0,0 +1,118 @@
+	// TODO(#12) Implement Authorization
+	ii, err := conv.InitiativeInvitationFromOAPI(request.Body)
+	if err != nil {
+		return nil, err
+	}
+	id, err := s.DB.CreateInitiativeInvitation(s.DB.NoTxn(ctx), ii)
+	if err != nil {
+		return nil, oapierr.Internal("failed to create initiative invitation", zap.Error(err))
+	}
+	if id != ii.ID {
+		return nil, oapierr.Internal(
+			"failed to create initiative invitation: ID mismatch",
+			zap.String("requested_id", string(ii.ID)),
+			zap.String("actual_id", string(id)),
+		)
+	}
+	return api.CreateInitiativeInvitation204Response{}, nil
+}
+
+// Deletes an initiative invitation by id
+// (DELETE /initiative-invitation/{id})
+func (s *Server) DeleteInitiativeInvitation(ctx context.Context, request api.DeleteInitiativeInvitationRequestObject) (api.DeleteInitiativeInvitationResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	err := s.DB.DeleteInitiativeInvitation(s.DB.NoTxn(ctx), pacta.InitiativeInvitationID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to delete initiative invitation", zap.Error(err))
+	}
+	return api.DeleteInitiativeInvitation204Response{}, nil
+}
+
+// Returns the initiative invitation from this id, if it exists
+// (GET /initiative-invitation/{id})
+func (s *Server) GetInitiativeInvitation(ctx context.Context, request api.GetInitiativeInvitationRequestObject) (api.GetInitiativeInvitationResponseObject, error) {
+	// TODO(#12) Implement Authorization
+	ii, err := s.DB.InitiativeInvitation(s.DB.NoTxn(ctx), pacta.InitiativeInvitationID(request.Id))
+	if err != nil {
+		return nil, oapierr.Internal("failed to retrieve initiative invitation", zap.Error(err))
+	}
+	result, err := conv.InitiativeInvitationToOAPI(ii)
+	if err != nil {
+		return nil, err
+	}
+	return api.GetInitiativeInvitation200JSONResponse(*result), nil
+}
+
+// Claims this initiative invitation, if it exists
+// (POST /initiative-invitation/{id})
+func (s *Server) ClaimInitiativeInvitation(ctx context.Context, request api.ClaimInitiativeInvitationRequestObject) (api.ClaimInitiativeInvitationResponseObject, error) {
+	userID, err := getUserID(ctx)
+	if err != nil {
+		return nil, err
+	}
+	err = s.DB.Transactional(ctx, func(tx db.Tx) error {
+		ii, err := s.DB.InitiativeInvitation(tx, pacta.InitiativeInvitationID(request.Id))
+		if err != nil {
+			return fmt.Errorf("looking up initiative invite: %w", err)
+		}
+		if !ii.UsedAt.IsZero() || ii.UsedBy != nil {
+			if ii.UsedBy != nil && ii.UsedBy.ID == userID {
+				// We don't return an error if the same user tries to claim the same invitation twice,
+				// which might happen by accident, but wouldn't impact the state of the initiative memberships.
+				// We may want to log this, though.
+				return nil
+			} else {
+				return fmt.Errorf("initiative is already used: %+v", ii)
```

Comment:
We should be using oapierr.BadRequest (or similar) for this. For this case in particular, we likely want a custom error code so we can show a specific error to the end user. Then you'd just need to update the error logic below to:

```go
if errors.Is(err, oapierr.Error{}) { // or something like that, could also make oapierr.Is(err) or similar
	return err
} else if err != nil {
	return oapierr.Internal(...)
}
```

You could even make a lil helper, like `return oapierr.OrInternal(err)`

Reply:
Done, but through different means, as an external var.